orderbook/src/orderbook.rs

```diff
@@ -48,6 +48,14 @@ impl OrderBook {
         self.orders.read().await.clone()
     }
 
+    pub async fn get_order_by_uid(&self, uid: OrderUid) -> Option<Order> {
+        let orders = self.get_orders().await;
+        orders
+            .into_iter()
+            .filter(|order| order.order_meta_data.uid == uid)
+            .last()
```

Comment:
Why last instead of first? I assume we want to guarantee that an orderUid can only be in the list once. Can we maybe enforce this at insertion time (and/or even using a more duplicate resistant datastructure like a HashSet or Map?) and then (debug) assert that there is only one match here.

Reply:
Yes now that we have the uid we can turn the vector into a hash map.

Reply:
I rebased on the latest main commit with the code containing the new hashmap structure.